endpoints/put_test.go

```diff
@@ -1109,3 +1141,22 @@ func (b *deadlineExceedingBackend) Put(ctx context.Context, key string, value st
 func newDeadlineExceededBackend() *deadlineExceedingBackend {
 	return &deadlineExceedingBackend{}
 }
+
+type countingCallsBackend struct {
+	GetCalls int
+	PutCalls int
+}
+
+func (b *countingCallsBackend) Get(ctx context.Context, key string) (string, error) {
+	b.GetCalls++
+	return "", nil
+}
+
+func (b *countingCallsBackend) Put(ctx context.Context, key string, value string, ttlSeconds int) error {
+	b.PutCalls++
+	return nil
+}
+
+func newCountingCallsBackend() *countingCallsBackend {
+	return &countingCallsBackend{}
+}
```

Comment:
This call counting is prone to race conditions due to the parallel processing in the code. This is also a case better handled by using testify's mock framework with call count expectations.

Switching to this fake backend will always return empty strings, but the memory backend that was used before I think would return real data. Is that important for these tests? Were we validating the response of these methods?

Reply:
Resolved in a meeting, @SyntaxNode commited the changes that fix this race condition and removes this part of the code no longer necessary